domain/application/service/application.go

```diff
@@ -353,6 +358,17 @@ func (s *ApplicationService) AddUnits(ctx context.Context, name string, units ..
 	return errors.Annotatef(err, "adding units to application %q", name)
 }
 
+// GetUnitUUID returns the UUID for the named unit
+func (s *ApplicationService) GetUnitUUID(ctx context.Context, unitName string) (coreunit.UUID, error) {
+	var result coreunit.UUID
+	err := s.st.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
```

Comment:
Do we need `RunAtomic()` here? This method seems so straightforward and without any logic whatsoever.

Reply:
We do because the state method is an atomic method. It's called in a few other places